[CI] ClusterDisruptionIT testAckedIndexing failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/262tw2urcmfja/tests/:server:internalClusterTest/org.elasticsearch.discovery.ClusterDisruptionIT/testAckedIndexing

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.discovery.ClusterDisruptionIT.testAckedIndexing" -Dtests.seed=F227DC2661BD3347 -Dtests.locale=sr-Latn-RS -Dtests.timezone=America/Halifax -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.discovery.ClusterDisruptionIT&tests.test=testAckedIndexing

**Failure excerpt:**
```
java.lang.AssertionError: failed to reach a stable cluster of [3] nodes. Tried via [node_t0]. last cluster state:
cluster uuid: -XwQ51BpTsyQMw_Wjp7XDg [committed: true]
version: 64
state uuid: XfzY1rwHSNWCym8FAHukTg
from_diff: false
meta data version: 16
   coordination_metadata:
      term: 2
      last_committed_config: VotingConfiguration{JuSgh8LOR9WQxzHHDqqlVQ,teMmveWdQ8eJomW0l7COzg,XZTfvsT_T8G2zfpKc6vMiQ}
      last_accepted_config: VotingConfiguration{JuSgh8LOR9WQxzHHDqqlVQ,teMmveWdQ8eJomW0l7COzg,XZTfvsT_T8G2zfpKc6vMiQ}
      voting tombstones: []
   [test/0e3yQZiiQryESs-x7Qtacw]: v[13], mv[4], sv[1], av[1]
      0: p_term [1], isa_ids [GgrbxbMaTNCpILvZl4Ep7A, vByWNTvzQHKjmYEG4ZNTeg]
      1: p_term [1], isa_ids [v_Fpb5_fRW-Di9kVj35n6g, geKyf7jqSVWDQpWqMVQTmw]
metadata customs:
   index-graveyard: IndexGraveyard[[]]   persistent_tasks: {"last_allocation_id":1,"tasks":[{"id":"health-node","task":{"health-node":{"params":{}}},"allocation_id":1,"assignment":{"executor_node":"JuSgh8LOR9WQxzHHDqqlVQ","explanation":""}}]}
nodes: 
   {node_t1}{XZTfvsT_T8G2zfpKc6vMiQ}{GEOyX6yBQK6q12UWfW01AA}{node_t1}{127.0.0.1}{127.0.0.1:19091}{cdfhilmrstw}
   {node_t2}{JuSgh8LOR9WQxzHHDqqlVQ}{Di5DKi8_Q3OesY4wpcnCfA}{node_t2}{127.0.0.1}{127.0.0.1:19093}{cdfhilmrstw}, master
   {node_t0}{teMmveWdQ8eJomW0l7COzg}{UqjXsHaTQKax1c3zc-P9iA}{node_t0}{127.0.0.1}{127.0.0.1:19092}{cdfhilmrstw}
routing_table (version 14):
-- index [[test/0e3yQZiiQryESs-x7Qtacw]]
----shard_id [test][0]
--------[test][0], node[teMmveWdQ8eJomW0l7COzg], [R], s[STARTED], a[id=vByWNTvzQHKjmYEG4ZNTeg], failed_attempts[0]
--------[test][0], node[JuSgh8LOR9WQxzHHDqqlVQ], [P], s[STARTED], a[id=GgrbxbMaTNCpILvZl4Ep7A], failed_attempts[0]
----shard_id [test][1]
--------[test][1], node[teMmveWdQ8eJomW0l7COzg], relocating [XZTfvsT_T8G2zfpKc6vMiQ], [P], s[RELOCATING], a[id=v_Fpb5_fRW-Di9kVj35n6g, rId=vG7OtoSySlafxcOzoHgVMA], failed_attempts[0], expected_shard_size[225]
--------[test][1], node[JuSgh8LOR9WQxzHHDqqlVQ], [R], s[STARTED], a[id=geKyf7jqSVWDQpWqMVQTmw], failed_attempts[0]

routing_nodes:
-----node_id[teMmveWdQ8eJomW0l7COzg][V]
--------[test][0], node[teMmveWdQ8eJomW0l7COzg], [R], s[STARTED], a[id=vByWNTvzQHKjmYEG4ZNTeg], failed_attempts[0]
--------[test][1], node[teMmveWdQ8eJomW0l7COzg], relocating [XZTfvsT_T8G2zfpKc6vMiQ], [P], s[RELOCATING], a[id=v_Fpb5_fRW-Di9kVj35n6g, rId=vG7OtoSySlafxcOzoHgVMA], failed_attempts[0], expected_shard_size[225]
-----node_id[JuSgh8LOR9WQxzHHDqqlVQ][V]
--------[test][0], node[JuSgh8LOR9WQxzHHDqqlVQ], [P], s[STARTED], a[id=GgrbxbMaTNCpILvZl4Ep7A], failed_attempts[0]
--------[test][1], node[JuSgh8LOR9WQxzHHDqqlVQ], [R], s[STARTED], a[id=geKyf7jqSVWDQpWqMVQTmw], failed_attempts[0]
-----node_id[XZTfvsT_T8G2zfpKc6vMiQ][V]
--------[test][1], node[XZTfvsT_T8G2zfpKc6vMiQ], relocating [teMmveWdQ8eJomW0l7COzg], [P], recovery_source[peer recovery], s[INITIALIZING], a[id=vG7OtoSySlafxcOzoHgVMA, rId=v_Fpb5_fRW-Di9kVj35n6g], failed_attempts[0], expected_shard_size[225]
---- unassigned
customs:
   health: org.elasticsearch.health.metadata.HealthMetadata@ff2c405a

  at org.junit.Assert.fail(Assert.java:88)
  at org.elasticsearch.test.ESIntegTestCase.ensureStableCluster(ESIntegTestCase.java:1295)
  at org.elasticsearch.discovery.ClusterDisruptionIT.testAckedIndexing(ClusterDisruptionIT.java:224)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/discovery/ClusterDisruptionIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/discovery/ClusterDisruptionIT.java
@@ -94,18 +94,11 @@ public class ClusterDisruptionIT extends AbstractDisruptionTestCase {
     }
 
     /**
-     * Test that we do not loose document whose indexing request was successful, under a randomly selected disruption scheme
+     * Test that we do not lose documents, indexed via requests that return success, under randomly selected disruption schemes.
      * We also collect &amp; report the type of indexing failures that occur.
      * <p>
-     * This test is a superset of tests run in the Jepsen test suite, with the exception of versioned updates
+     * This test is a superset of tests run in the Jepsen test suite, with the exception of versioned updates.
      */
-    @TestIssueLogging(
-        value = "_root:DEBUG,org.elasticsearch.action.bulk:TRACE,org.elasticsearch.action.get:TRACE,"
-            + "org.elasticsearch.discovery:TRACE,org.elasticsearch.action.support.replication:TRACE,"
-            + "org.elasticsearch.cluster.service:TRACE,org.elasticsearch.indices.recovery:TRACE,"
-            + "org.elasticsearch.indices.cluster:TRACE,org.elasticsearch.index.shard:TRACE",
-        issueUrl = "https://github.com/elastic/elasticsearch/issues/41068"
-    )
     public void testAckedIndexing() throws Exception {
 
         final int seconds = (TEST_NIGHTLY && rarely()) == false ? 1 : 5;

--- a/server/src/internalClusterTest/java/org/elasticsearch/discovery/ClusterDisruptionIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/discovery/ClusterDisruptionIT.java
@@ -40,6 +40,7 @@ import org.elasticsearch.test.disruption.NetworkDisruption;
 import org.elasticsearch.test.disruption.NetworkDisruption.Bridge;
 import org.elasticsearch.test.disruption.NetworkDisruption.TwoPartitions;
 import org.elasticsearch.test.disruption.ServiceDisruptionScheme;
+import org.elasticsearch.test.junit.annotations.TestLogging;
 import org.elasticsearch.test.transport.MockTransportService;
 import org.elasticsearch.xcontent.XContentType;
 
@@ -98,6 +99,13 @@ public class ClusterDisruptionIT extends AbstractDisruptionTestCase {
      * <p>
      * This test is a superset of tests run in the Jepsen test suite, with the exception of versioned updates.
      */
+    @TestLogging(
+        value = "_root:DEBUG,org.elasticsearch.action.bulk:TRACE,org.elasticsearch.action.get:TRACE,"
+            + "org.elasticsearch.discovery:TRACE,org.elasticsearch.action.support.replication:TRACE,"
+            + "org.elasticsearch.cluster.service:TRACE,org.elasticsearch.indices.recovery:TRACE,"
+            + "org.elasticsearch.indices.cluster:TRACE,org.elasticsearch.index.shard:TRACE",
+        reason = "Past failures have required a lot of additional logging to debug"
+    )
     public void testAckedIndexing() throws Exception {
 
         final int seconds = (TEST_NIGHTLY && rarely()) == false ? 1 : 5;

--- a/server/src/internalClusterTest/java/org/elasticsearch/discovery/ClusterDisruptionIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/discovery/ClusterDisruptionIT.java
@@ -40,7 +40,6 @@ import org.elasticsearch.test.disruption.NetworkDisruption;
 import org.elasticsearch.test.disruption.NetworkDisruption.Bridge;
 import org.elasticsearch.test.disruption.NetworkDisruption.TwoPartitions;
 import org.elasticsearch.test.disruption.ServiceDisruptionScheme;
-import org.elasticsearch.test.junit.annotations.TestIssueLogging;
 import org.elasticsearch.test.transport.MockTransportService;
 import org.elasticsearch.xcontent.XContentType;
 

--- a/server/src/test/java/org/elasticsearch/discovery/AbstractDisruptionTestCase.java
+++ b/server/src/test/java/org/elasticsearch/discovery/AbstractDisruptionTestCase.java
@@ -13,6 +13,7 @@ import org.elasticsearch.cluster.block.ClusterBlock;
 import org.elasticsearch.cluster.block.ClusterBlockLevel;
 import org.elasticsearch.cluster.coordination.Coordinator;
 import org.elasticsearch.cluster.coordination.FollowersChecker;
+import org.elasticsearch.cluster.coordination.LagDetector;
 import org.elasticsearch.cluster.coordination.LeaderChecker;
 import org.elasticsearch.cluster.node.DiscoveryNodes;
 import org.elasticsearch.common.settings.Settings;
@@ -116,6 +117,7 @@ public abstract class AbstractDisruptionTestCase extends ESIntegTestCase {
         .put(FollowersChecker.FOLLOWER_CHECK_TIMEOUT_SETTING.getKey(), "5s") // for hitting simulated network failures quickly
         .put(FollowersChecker.FOLLOWER_CHECK_RETRY_COUNT_SETTING.getKey(), 1) // for hitting simulated network failures quickly
         .put(Coordinator.PUBLISH_TIMEOUT_SETTING.getKey(), "5s") // <-- for hitting simulated network failures quickly
+        .put(LagDetector.CLUSTER_FOLLOWER_LAG_TIMEOUT_SETTING.getKey(), "5s") // remove lagging nodes quickly so they can rejoin
         .put(TransportSettings.CONNECT_TIMEOUT.getKey(), "10s") // Network delay disruption waits for the min between this
         // value and the time of disruption and does not recover immediately
         // when disruption is stop. We should make sure we recover faster

